File: basicAlgorithms/selection_sort.cpp
#include <iostream>
using namespace std;
void selection_sort(int arr[], int len)
{
    int smallest;
    int j = 0;
    int smallest_index = 0;
    for (int i = 1; i < len; i++)
    {
        smallest = arr[i - 1];
        for (j = i; j < len; j++)
        {
            if (smallest > arr[j])
            {
                smallest = arr[j];
                smallest_index = j;
            }
        }
        int temp;
        temp = arr[i - 1];
        arr[i - 1] = arr[smallest_index];
        arr[smallest_index] = temp;
        for (int i = 0; i < len; i++)
        {
            cout << arr[i] << "\t";
        }
        cout << endl;
    }
}
int main(int argc, char const *argv[])
{
    int a[] = {54, 1, 2, 4, 3, 2, 4};
    selection_sort(a, 7);
}

File: basicAlgorithms/selection_sort.py
def selectionSort(array):
    smallest_index = 0
    for i in range(1, len(array)):
        j = i
        smallest = array[i-1]
        smallest_index = i-1
        while(j < len(array)):
            if(smallest > array[j]):
                smallest = array[j]
                smallest_index = j
            j += 1
        array[i-1], array[smallest_index] = array[smallest_index], array[i-1]
        print(smallest_index)
        for i in array:
            print(i, end="\t")
        print()

    return array


arr = [54, 1, 2, 4, 3, 2, 4]
selectionSort(arr)
for i in arr:
    print(i, end=" ")
